API/test/app.test.js

```diff
@@ -46,10 +54,81 @@ describe('App', () => {
       chai.request(app)
         .get('/api/v1')
         .end((error, response) => {
+          assert.strictEqual(response.status, 200);
+          assert.hasAllKeys(response.body, ['status', 'message']);
+          done();
+        });
+    });
+  });
+
+  describe('/POST /api/v1/auth/signup', () => {
+    it('should not signup if payload has additional properties', (done) => {
+      chai.request(app)
+        .post('/api/v1/auth/signup')
+        .send({
+          firstName,
+          lastName,
+          email,
+          password,
+          additional: ''
+        })
+        .end((error, response) => {
+          assert.strictEqual(response.status, 422);
+          assert.hasAllKeys(response.body, ['status', 'message']);
+          done();
+        });
+    });
+
+    it('should not signup if any required input is invalid', (done) => {
+      chai.request(app)
+        .post('/api/v1/auth/signup')
+        .send({
+          firstName,
+          lastName,
+          email,
+          password: emptyPassword
+        })
+        .end((error, response) => {
+          assert.strictEqual(response.status, 422);
           assert.hasAllKeys(response.body, ['status', 'message']);
           done();
         });
     });
+
+    it('should not signup if the email has already been used', (done) => {
+      UsersDBQueries.createUser(firstName, lastName, email, password)
+        .then(() => {
+          chai.request(app)
+            .post('/api/v1/auth/signup')
+            .send({
+              firstName,
+              lastName,
+              email,
+              password
+            })
+            .end((error, response) => {
+              assert.strictEqual(response.status, 422);
+              assert.hasAllKeys(response.body, ['status', 'message']);
+              done();
+            });
+        });
+    });
+
+    it('should signup', (done) => {
+      chai.request(app)
+        .post('/api/v1/auth/signup')
+        .send({
+          firstName,
+          lastName,
+          email,
+          password
+        })
+        .end((error, response) => {
+          assert.strictEqual(response.status, 201);
+          assert.hasAllKeys(response.body, ['status', 'message', 'data']);
```

Comment:
Perhaps you might want to assert that the values are what you expected alongside what you have now.

Reply:
Ok